apps/merkle_patricia_tree/lib/hex_prefix.ex

```diff
@@ -0,0 +1,108 @@
+defmodule MerklePatriciaTree.HexPrefix do
+  @moduledoc """
+  Elixir implementation of Ethereum's Hex Prefix encoding
+  The encoding's specification can be found in [the yellow paper](http://yellowpaper.io/) under Appendix C.
+  """
+
+  @type t :: {[0x0..0xF], boolean()}
+
+  @doc """
+  Encodes a list of nibbles using hex-prefix notation.
+
+  ## Examples
+
+    iex> MerklePatriciaTree.HexPrefix.encode({[0xa, 0xb, 0xc, 0xd], false})
+    <<0, 171, 205>>
+
+    iex> MerklePatriciaTree.HexPrefix.encode({[0xa, 0xb, 0xc, 0xd], true})
+    <<32, 171, 205>>
+
+    iex> MerklePatriciaTree.HexPrefix.encode({[0x09, 0xa, 0xb, 0xc, 0xd], false})
+    <<25, 171, 205>>
+
+    iex> MerklePatriciaTree.HexPrefix.encode({[0x09, 0xa, 0xb, 0xc, 0xd], true})
+    <<57, 171, 205>>
+
+    iex> MerklePatriciaTree.HexPrefix.encode({[ 1, 2, 3, 4, 5 ], false})
+    <<0x11, 0x23, 0x45>>
+
+    iex> MerklePatriciaTree.HexPrefix.encode({[ 0, 1, 2, 3, 4, 5 ], false})
+    <<0x00, 0x01, 0x23, 0x45>>
+
+    iex> MerklePatriciaTree.HexPrefix.encode({[ 0, 15, 1, 12, 11, 8 ], true})
+    <<0x20, 0x0f, 0x1c, 0xb8>>
+
+    iex> MerklePatriciaTree.HexPrefix.encode({[ 15, 1, 12, 11, 8 ], true})
+    <<0x3f, 0x1c, 0xb8>>
+  """
+  @spec encode(__MODULE__.t()) :: binary()
```

Comment:
`__MODULE__.t()` can be replaces with t()

Reply:
thx, didn't notice that